packages/xrpl/src/sugar/submit.ts

```diff
@@ -95,23 +107,25 @@ async function submitAndWait(
 // Helper functions
 
 // Encodes and submits a signed transaction.
+// eslint-disable-next-line max-params -- All params are required
 async function submitRequest(
   client: Client,
   signedTransaction: Transaction | string,
   failHard = false,
+  definitions?: XrplDefinitions,
 ): Promise<SubmitResponse> {
-  if (!isSigned(signedTransaction)) {
+  if (!isSigned(signedTransaction, definitions)) {
     throw new ValidationError('Transaction must be signed')
   }
 
   const signedTxEncoded =
     typeof signedTransaction === 'string'
       ? signedTransaction
-      : encode(signedTransaction)
+      : encode(signedTransaction, definitions)
   const request: SubmitRequest = {
     command: 'submit',
     tx_blob: signedTxEncoded,
-    fail_hard: isAccountDelete(signedTransaction) || failHard,
+    fail_hard: isAccountDelete(signedTransaction, definitions) || failHard,
```

Comment:
ditto

Reply:
Same issue - without the optional definitions it's possible to not have enough information to decode a new transaction type.